test/integration/sorted-set.test.ts

```diff
@@ -187,174 +186,434 @@ describe('Integration tests for sorted set operations', () => {
-    });
+      it('should fetch only the specified range if start index is specified', async () => {
+        const response = await Momento.sortedSetFetchByIndex(
+          IntegrationTestCacheName,
+          sortedSetName,
+          {
+            startIndex: 4,
+          }
+        );
+
+        expect(response).toBeInstanceOf(CacheSortedSetFetch.Hit);
+        const hitResponse = response as CacheSortedSetFetch.Hit;
+        expect(hitResponse.valueArray()).toEqual([
+          {value: 'bam', score: 1000},
+          {value: 'burrito', score: 9000},
+          {value: 'taco', score: 90210},
+          {value: 'jalapeno', score: 1_000_000},
+        ]);
+      });
 
-    it('should provide value accessors for bytes fields with dictionaryFetch', async () => {
-      const textEncoder = new TextEncoder();
+      it('should fetch only the specified range if end index is specified', async () => {
+        const response = await Momento.sortedSetFetchByIndex(
+          IntegrationTestCacheName,
+          sortedSetName,
+          {
+            endIndex: 3,
+          }
+        );
+
+        expect(response).toBeInstanceOf(CacheSortedSetFetch.Hit);
+        const hitResponse = response as CacheSortedSetFetch.Hit;
+        expect(hitResponse.valueArray()).toEqual([
+          {value: 'foo', score: 1},
+          {value: 'bar', score: 2},
+          {value: 'baz', score: 42},
+        ]);
+      });
 
-      const dictionaryName = v4();
-      const field1 = textEncoder.encode(v4());
-      const value1 = v4();
-      const field2 = textEncoder.encode(v4());
-      const value2 = v4();
+      it('should fetch only the specified range if both start and end index are specified', async () => {
+        const response = await Momento.sortedSetFetchByIndex(
+          IntegrationTestCacheName,
+          sortedSetName,
+          {
+            startIndex: 1,
+            endIndex: 5,
+          }
+        );
+
+        expect(response).toBeInstanceOf(CacheSortedSetFetch.Hit);
+        const hitResponse = response as CacheSortedSetFetch.Hit;
+        expect(hitResponse.valueArray()).toEqual([
+          {value: 'bar', score: 2},
+          {value: 'baz', score: 42},
+          {value: 'habanero', score: 68},
+          {value: 'bam', score: 1000},
+        ]);
+      });
 
-      await Momento.dictionarySetFields(
-        IntegrationTestCacheName,
-        dictionaryName,
-        new Map([
-          [field1, value1],
-          [field2, value2],
-        ])
-      );
+      it('should return an empty list if start index is out of bounds', async () => {
+        const response = await Momento.sortedSetFetchByIndex(
+          IntegrationTestCacheName,
+          sortedSetName,
+          {
+            startIndex: 10,
+          }
+        );
+
+        expect(response).toBeInstanceOf(CacheSortedSetFetch.Hit);
+        const hitResponse = response as CacheSortedSetFetch.Hit;
+        expect(hitResponse.valueArray()).toEqual([]);
+      });
```

Comment:
@wandaitzuchen - I don't need you to review this whole PR, just this one test.

This test is failing; we are getting back a `.missing` value on the response object.  I feel like that is not the right behavior?  I feel like this should return a Hit with an empty list of values.  WDYT?

Reply:
Actually in the backend we do return a miss if we got an empty list from redis (redis response does not differentiate between sorted set not exist or fetching empty list from existing sorted set). Unlike sorted set get rank or sorted set get score, where `_SortedSetFound` is returned to include `ECacheResult`, sorted set fetch return found only when there's elements in the response.  

I agree that we should unify the behavior and return a found with empty list if the set does exist.